variable.tf

```diff
@@ -185,4 +185,20 @@ variable "tls_security_policy" {
     condition     = contains(["Policy-Min-TLS-1-2-2019-07", "Policy-Min-TLS-1-0-2019-07"], var.tls_security_policy)
     error_message = "Valid values for tls_security_policy are: (Policy-Min-TLS-1-2-2019-07, Policy-Min-TLS-1-0-2019-07)."
   }
+}
+
+variable "iam_users" {
+  type = map(list(object({
+    http_methods = list(string)
+    http_paths   = list(string)
+  })))
+  description = "IAM users to create and their allowed HTTP paths and methods"
+
+  validation {
+    condition = alltrue(
+      [for user in var.iam_users : alltrue(
+        [for policy in user : alltrue(
+          [for method in policy.http_methods : contains(["Post", "Get", "Put", "Delete", "Head", "Patch", "*"], method)])])])
```

Comment:
Suggest to make it case insensitive, very common for someone to use POST,GET etc.